core/embed/rust/src/micropython/list.rs

```diff
@@ -17,6 +17,34 @@ impl List {
         })
     }
 
+    pub fn with_capacity(capacity: usize) -> Result<Gc<Self>, Error> {
+        // EXCEPTION: Will raise if allocation fails.
+        catch_exception(|| unsafe {
+            let list: *mut Self = ffi::mp_obj_new_list(capacity, ptr::null_mut())
+                .as_ptr()
+                .cast();
+            // By default, the new list will have its len set to n. We want to preallocate
+            // to a specific size and then use append() to add items, so we reset len to 0.
+            (*list).len = 0;
```

Comment:
`ffi::mp_obj_list_set_len` might be cleaner?

Reply:
good find! i didn't notice that exists.

great comment too:
```
    // trust that the caller knows what it's doing
```

Reply:
resolved in [d12f8c2](https://github.com/trezor/trezor-firmware/pull/2235/commits/d12f8c217e34c8875f9d15929055227be79bd7ae)